graphql/src/datasources/es/api/getQueryResults.ts

```diff
@@ -0,0 +1,315 @@
+    ...buildArrayFilter(
+      'venue.location.administrativeDivisions.id.keyword',
+      administrativeDivisionIds ?? []
+    ),
+    ...(ontologyTreeIdOrSets ?? []).flatMap((ontologyTreeIdOrSet) =>
+      buildArrayFilter(
+        'links.raw_data.ontologytree_ids_enriched.id',
+        ontologyTreeIdOrSet ?? []
+      )
+    ),
+    ...(ontologyWordIdOrSets ?? []).flatMap((ontologyWordIdOrSet) =>
+      buildArrayFilter(
+        'links.raw_data.ontologyword_ids_enriched.id',
+        ontologyWordIdOrSet ?? []
+      )
+    ),
+    ...buildArrayFilter(
+      'venue.serviceOwner.providerType.keyword',
+      providerTypes ?? []
+    ),
+    ...buildArrayFilter(
+      'venue.serviceOwner.type.keyword',
+      serviceOwnerTypes ?? []
+    ),
+    ...buildArrayFilter('venue.targetGroups.keyword', targetGroups ?? []),
+    ...(mustHaveReservableResource
+      ? [buildMustHaveReservableResourceFilter()]
+      : []),
+    ...(finalOpenAt
+      ? [
+          {
+            term: {
+              'venue.openingHours.openRanges': finalOpenAt,
+            },
+          },
+        ]
+      : []),
+  ];
+
+  if (filters.length > 0) {
+    query.query.bool.minimum_should_match = 1;
+    query.query.bool.filter = filters;
+  }
+
+  if (es_index === ES_EVENT_INDEX || es_index === ES_LOCATION_INDEX) {
+    const searchResultField = ElasticSearchIndexToSearchResultField[es_index];
+    const language = languages[0] ?? DEFAULT_ELASTIC_LANGUAGE;
+
+    if (isDefined(orderByDistance)) {
+      query.sort = {
+        _geo_distance: {
+          location: {
+            lat: orderByDistance.latitude,
+            lon: orderByDistance.longitude,
+          },
+          order: orderByDistance.order === 'DESCENDING' ? 'desc' : 'asc',
+          ignore_unmapped: true,
+        },
+      };
+    } else if (isDefined(orderByName)) {
+      query.sort = {
+        [`${searchResultField}.name.${language}.keyword`]: {
+          order: orderByName.order === 'DESCENDING' ? 'desc' : 'asc',
+          missing: '_last',
+        },
+      };
+    } else if (
+      isDefined(orderByAccessibilityProfile) &&
+      searchResultField === VENUE_SEARCH_RESULT_FIELD
+    ) {
+      query.sort = [
+        // Primary sort field (chosen accessibility profile's shortcoming count)
+        {
+          'venue.accessibility.shortcomings.count': {
+            order: 'asc',
+            nested: {
+              path: 'venue.accessibility.shortcomings',
+              filter: {
+                term: {
+                  'venue.accessibility.shortcomings.profile':
+                    orderByAccessibilityProfile,
+                },
+              },
+              max_children: 1,
+            },
+            missing: '_last',
+          },
+        },
+        // Secondary sort field (venue's name)
+        {
+          [`${searchResultField}.name.${language}.keyword`]: {
+            order: 'asc',
+            missing: '_last',
+          },
+        },
+      ];
+    }
+  }
```

Comment:
This part needs to be refactored. It's so hard to read that I don't understand what it is trying to achieve...

Reply:
With a second look, it seems to be all some sorting stuff. Should be moved tp some sorting-getter-function.